RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1481 @@
+            //Loop over each channel, first compute soiSamples for all channels
+            for (auto channel : uniform_group_elements_y(acc, group, nchannels)) {
+              auto const gch = channel.global;
+              if (gch >= nchannels)
+                return;
+
+              for (auto i_sample : independent_group_elements_x(acc, nsamplesForCompute)) {
+                auto const sampleWithinWindow = i_sample;
+                auto const sample = i_sample + startingSample;
+                // initialize
+                if (sampleWithinWindow == 0) {
+                  soiSamples[gch] = -1;
+                }
+                if (gch < f01HEDigis.size()) {
+                  // NOTE: assume that soi is high only for a single guy!
+                  //   which must be the case. cpu version does not check for that
+                  //   if that is not the case, we will see that with cuda mmecheck
+                  auto const soibit = soibit_for_sample<Flavor1>(&(f01HEDigis.data()[gch][0]), sample);
+                  if (soibit == 1)
+                    soiSamples[gch] = sampleWithinWindow;
+                } else if (gch >= nchannelsf015) {
+                  auto const soibit = soibit_for_sample<Flavor3>(&(f3HBDigis.data()[gch - nchannelsf015][0]), sample);
+                  if (soibit == 1)
+                    soiSamples[gch] = sampleWithinWindow;
+                }
+              }
+            }
+            alpaka::syncBlockThreads(acc);
+
+            //Loop over each channel, compute input for multifit
+            for (auto channel : uniform_group_elements_y(acc, group, nchannels)) {
+              auto const gch = channel.global;
+              auto const lch = channel.local;
+
+              if (gch >= nchannels)
+                return;
+
+              //Loop over sample
+              for (auto i_sample : independent_group_elements_x(acc, nsamplesForCompute)) {
+                auto const sampleWithinWindow = i_sample;
+                auto const sample = i_sample + startingSample;
+                auto const linearThPerBlock = i_sample + channel.local * nsamplesForCompute;
+
+                // initialize all output buffers
+                if (sampleWithinWindow == 0) {
+                  outputGPU.did()[gch] = 0;
+                  outputGPU.energyM0()[gch] = 0;
+                  outputGPU.timeM0()[gch] = 0;
+                  outputGPU.energy()[gch] = 0;
+                  outputGPU.chi2()[gch] = 0;
+                }
+
+                // offset output
+                auto* amplitudesForChannel = amplitudes + nsamplesForCompute * gch;
+                auto* noiseTermsForChannel = noiseTerms + nsamplesForCompute * gch;
+                auto* electronicNoiseTermsForChannel = electronicNoiseTerms + nsamplesForCompute * gch;
+
+                // configure shared mem
+                float* smem = alpaka::getDynSharedMem<float>(acc);
+                float* shrEnergyM0PerTS = reinterpret_cast<float*>(smem);
```

Comment:
This `reinterpret_cast` seems unnecessary. How about
```suggestion
                float* smem = alpaka::getDynSharedMem<float>(acc);
                float* shrEnergyM0PerTS = smem;
```
or
```suggestion
                float* shrEnergyM0PerTS = alpaka::getDynSharedMem<float>(acc);
```
?

Reply:
done in https://github.com/cms-sw/cmssw/commit/98367cbd606553c45f30a100a438bedbe880ae55